langchain4j/src/main/java/dev/langchain4j/service/AiServiceStreamingResponseHandler.java

```diff
@@ -193,6 +201,29 @@ public void onPartialThinking(PartialThinking partialThinking, PartialThinkingCo
         }
     }
 
+    @Override
+    public void onPartialToolCall(PartialToolCall partialToolCall) {
+        if (partialToolCallHandler != null) {
+            partialToolCallHandler.accept(partialToolCall);
+            return;
+        }
+        if (partialToolCallWithContextHandler != null) {
```

Comment:
```suggestion
        else if (partialToolCallWithContextHandler != null) {
```

we don't need to call both handlers

Reply:
I understand that we validate that only one of consumers is configured when creating `TokenStream`, but it would be nice to keep this explicit here as well